src/gui/codeeditors/qgscodeeditor.cpp

```diff
@@ -214,30 +217,69 @@ void QgsCodeEditor::keyPressEvent( QKeyEvent *event )
 
 void QgsCodeEditor::contextMenuEvent( QContextMenuEvent *event )
 {
-  if ( mMode != QgsCodeEditor::Mode::CommandInput )
+  switch ( mMode )
   {
-    QsciScintilla::contextMenuEvent( event );
-    return;
-  }
+    case Mode::ScriptEditor:
+    {
+      QMenu *menu = createStandardContextMenu();
+      menu->setAttribute( Qt::WA_DeleteOnClose );
+
+      if ( ( languageCapabilities() & Qgis::ScriptLanguageCapability::Reformat ) ||
+           ( languageCapabilities() & Qgis::ScriptLanguageCapability::CheckSyntax ) )
+      {
+        menu->addSeparator();
+      }
+
+      if ( languageCapabilities() & Qgis::ScriptLanguageCapability::Reformat )
+      {
+        QAction *reformatAction = new QAction( tr( "Reformat Code" ), menu );
+        reformatAction->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "console/iconFormatCode.svg" ) ) );
+        reformatAction->setEnabled( !isReadOnly() );
+        connect( reformatAction, &QAction::triggered, this, &QgsCodeEditor::reformatCode );
+        menu->addAction( reformatAction );
+      }
+
+      if ( languageCapabilities() & Qgis::ScriptLanguageCapability::Reformat )
+      {
+        QAction *syntaxCheckAction = new QAction( tr( "Check Syntax" ), menu );
+        syntaxCheckAction->setIcon( QgsApplication::getThemeIcon( QStringLiteral( "console/iconSyntaxErrorConsole.svg" ) ) );
+        connect( syntaxCheckAction, &QAction::triggered, this, &QgsCodeEditor::checkSyntax );
+        menu->addAction( syntaxCheckAction );
```

Comment:
I wonder if we could define a standard set of QAction for the QgsCodeEditor (e.g. format, toggle comment, checkSyntax, save etc...) that could then be added to any QToolbar or contextMenu.

Ideally, it would also be nice to define keyboard Shortcuts for these actions.

Reply:
That's on my long term plans 😁 I'm thinking of making it so you can associate a toolbar with an editor and it'll get auto populated with all the standard actions. Basically I want it to be super easy to create a complete script editor for any supported language, not just python...